extensions/notification/configManager/MongoConfigManager.js

```diff
@@ -94,46 +95,42 @@ class MongoConfigManager extends BaseConfigManager {
      * and retreives the metastore collection
      * @param {Function} cb callback
      * @returns {undefined}
-     */
-    _setupMongoClient(cb) {
-        const mongoUrl = constructConnectionString(this._mongoConfig);
-        MongoClient.connect(mongoUrl, {
-            replicaSet: this._mongoConfig.replicaSet,
-            useNewUrlParser: true,
-        },
-        (err, client) => {
-            if (err) {
-                this._logger.error('Could not connect to MongoDB', {
+    */
+    async _setupMongoClient(cb) {
+    const mongoUrl = constructConnectionString(this._mongoConfig);
+        try {
+            const client = await new MongoClient(mongoUrl, {
+                replicaSet: this._mongoConfig.replicaSet,
+                useNewUrlParser: true,
+                readPreference: this._mongoConfig.readPreference,
+            }).connect();
+            this._logger.debug('Connected to MongoDB', {
+                method: 'MongoConfigManager._setupMongoClient',
+            });
+
+            this._mongoClient = client.db(this._mongoConfig.database, {
+                ignoreUndefined: true,
+            });
+            this._metastore = this._mongoClient.collection(this._bucketMetastore);
+
+            try {
+                this._mongoVersion = await promisify(getMongoVersion)(this._mongoClient);
+            } catch (err) {
+                this._logger.error('Could not get MongoDB version', {
                     method: 'MongoConfigManager._setupMongoClient',
                     error: err.message,
                 });
                 return cb(err);
             }
```

Comment:
do not nest `try` blocks, terminate the "outer" block before starting this one. This improves readability, as there is no need to go back and forth.